MonoGame.Framework/Content/LzxDecoder.cs

```diff
@@ -1,780 +0,0 @@
- */
-#endregion
-#region MS-PL License
-/* 
- * MICROSOFT PUBLIC LICENSE
- * This source code is subject to the terms of the Microsoft Public License (Ms-PL). 
- *  
- * Redistribution and use in source and binary forms, with or without modification, 
- * is permitted provided that redistributions of the source code retain the above 
- * copyright notices and this file header. 
- *  
- * Additional copyright notices should be appended to the list above. 
- * 
- * For details, see <http://www.opensource.org/licenses/ms-pl.html>. 
- */
-#endregion
-/*
- * This derived work is recognized by Stuart Caie and is authorized to adapt
- * any changes made to lzxd.c in his libmspack library and will still retain
- * this dual licensing scheme. Big thanks to Stuart Caie!
- * 
- * DETAILS
- * This file is a pure C# port of the lzxd.c file from libmspack, with minor
- * changes towards the decompression of XNB files. The original decompression
- * software of LZX encoded data was written by Suart Caie in his
- * libmspack/cabextract projects, which can be located at 
- * http://http://www.cabextract.org.uk/
- */
-#endregion
-
-using System;
-using System.Diagnostics;
-
-namespace Microsoft.Xna.Framework.Content
-{
-	using System.IO;
-	
-	class LzxDecoder
-	{
-		public static uint[] position_base = null;
-		public static byte[] extra_bits = null;
-		
-		private LzxState m_state;
-		
-		public LzxDecoder (int window)
-		{
-			uint wndsize = (uint)(1 << window);
-			int posn_slots;
-			
-			// setup proper exception
-			if(window < 15 || window > 21) throw new UnsupportedWindowSizeRange();
-			
-			// let's initialise our state
-			m_state = new LzxState();
-			m_state.actual_size = 0;
-			m_state.window = new byte[wndsize];
-			for(int i = 0; i < wndsize; i++) m_state.window[i] = 0xDC;
-			m_state.actual_size = wndsize;
-			m_state.window_size = wndsize;
-			m_state.window_posn = 0;
-			
-			/* initialize static tables */
-			if(extra_bits == null)
-			{
-				extra_bits = new byte[52];
-				for(int i = 0, j = 0; i <= 50; i += 2)
-				{
-					extra_bits[i] = extra_bits[i+1] = (byte)j;
-					if ((i != 0) && (j < 17)) j++;
-				}
-			}
-			if(position_base == null)
-			{
-				position_base = new uint[51];
-				for(int i = 0, j = 0; i <= 50; i++)
-				{
-					position_base[i] = (uint)j;
-					j += 1 << extra_bits[i];
-				}
-			}
-			
-			/* calculate required position slots */
-			if(window == 20) posn_slots = 42;
-			else if(window == 21) posn_slots = 50;
-			else posn_slots = window << 1;
-			
-			m_state.R0 = m_state.R1 = m_state.R2 = 1;
-			m_state.main_elements = (ushort)(LzxConstants.NUM_CHARS + (posn_slots << 3));
-			m_state.header_read = 0;
-			m_state.frames_read = 0;
-			m_state.block_remaining = 0;
-			m_state.block_type = LzxConstants.BLOCKTYPE.INVALID;
-			m_state.intel_curpos = 0;
-			m_state.intel_started = 0;
-			
-			// yo dawg i herd u liek arrays so we put arrays in ur arrays so u can array while u array
```

Comment:
Double check the dual licenses allow modification of this line. If they do not then it might be a legal requirement to keep this 2008 meme in the source code. This opinion isn't legal advise, so please seek appropriate legal assistance and clarification.